src/libstd/io/mod.rs

```diff
@@ -589,6 +1050,23 @@ pub trait BufRead: Read {
     /// return an error if the read bytes are not valid UTF-8. If an I/O error
     /// is encountered then `buf` may contain some bytes already read in the
     /// event that all data read so far was valid UTF-8.
+    ///
+    /// # Examples
+    ///
+    /// A locked standard input implements `BufRead`:
+    ///
+    /// ```
+    /// use std::io;
+    /// use std::io::BufRead;
+    ///
+    /// let stdin = io::stdin();
+    /// let mut stdin = stdin.lock();
+    /// let mut buffer = String::new();
+    ///
+    /// stdin.read_line(&mut buffer).unwrap();
```

Comment:
can you expand on `truncate` here? It's not mentioned at all in the docs of this function, and this feels like something i should expand in text as well as in the sample